### PR TITLE
Add Support for Resource Groups Tagging API service

### LINKED ISSUE #1924: Add Support for ResourceTagging Api
## Feature Request

Data Sources would benefit if it was possible to filter using the resource tags.

- https://aws.amazon.com/blogs/aws/new-aws-resource-tagging-api/
- http://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/Welcome.html
- http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-aws-service-namespaces

for example, get load balancer arn by tag:

```bash
aws resourcegroupstaggingapi get-resources --resource-type-filters "elasticloadbalancing" --tag-filters Key="elasticbeanstalk:environment-name",Values="my-app" --tags-per-page 100 --query "ResourceTagMappingList[].ResourceARN" --output text
```

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX
```

### CODE REVIEW

**Comment by DrFaust92** [comment]
wip- needs docs and more tests

**Comment by DrFaust92** [comment]
test and docs added, resource type filter missing.

**Comment by DrFaust92** [comment]
Ready

File: aws/data_source_aws_resourcegroupstaggingapi_resources.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+		input.TagFilters = expandAwsResourceGroupsTaggingApiTagFilters(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("resource_type_filter"); ok && v.(*schema.Set).Len() > 0 {
+		input.ResourceTypeFilters = expandStringSet(v.(*schema.Set))
+	}
+
+	var taggings []*resourcegroupstaggingapi.ResourceTagMapping
+
+	err := conn.GetResourcesPages(input, func(page *resourcegroupstaggingapi.GetResourcesOutput, lastPage bool) bool {
```
Comment (ewbankkit) [inline aws/data_source_aws_resourcegroupstaggingapi_resources.go:113]: Need to better handle the case that `lastPage == false`.

File: aws/data_source_aws_resourcegroupstaggingapi_resources.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	return result
+}
+
+func flattenAwsResourceGroupsTaggingApiResourcesTagMappingList(list []*resourcegroupstaggingapi.ResourceTagMapping) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(list))
+
+	for _, i := range list {
+		l := map[string]interface{}{
+			"resource_arn": aws.StringValue(i.ResourceARN),
+			"tags":         keyvaluetags.ResourcegroupstaggingapiKeyValueTags(i.Tags).IgnoreAws().Map(),
```
Comment (ewbankkit) [inline aws/data_source_aws_resourcegroupstaggingapi_resources.go:154]: Will need to add support for `.IgnoreConfig(ignoreTagsConfig)`.

**Comment by DrFaust92** [comment]
changing this to scope this only for dep and service items. splitting the actual data source to a different PR.

**Pull request merged**